### PR TITLE
TST: Series construction from ExtensionDtype scalar

### LINKED ISSUE #28401: BUG: Constructing a Series from a scalar generally doesn't work for extension types
#### Code Sample, a copy-pastable example if possible
Construction without specifying a dtype results in `object` dtype for `Interval` and `Period`:
```python-traceback
In [2]: pd.Series(pd.Interval(0, 1), index=range(3))
Out[2]: 
0    (0, 1]
1    (0, 1]
2    (0, 1]
dtype: object

In [3]: pd.Series(pd.Period("2019Q1", freq="Q"), index=range(3))
Out[3]: 
0    2019Q1
1    2019Q1
2    2019Q1
dtype: object
```
This looks okay for a tz-aware `Timestamp`:
```python-traceback
In [4]: pd.Series(pd.Timestamp("2019", tz="US/Eastern"), index=range(3))
Out[4]: 
0   2019-01-01 00:00:00-05:00
1   2019-01-01 00:00:00-05:00
2   2019-01-01 00:00:00-05:00
dtype: datetime64[ns, US/Eastern]
```

Specifying a dtype raises for `Interval`, `Period` and tz-aware `Timestamp`:
```python-traceback
In [5]: pd.Series(pd.Interval(0, 1), index=range(3), dtype=pd.IntervalDtype("int64"))
---------------------------------------------------------------------------
TypeError: IntervalArray(...) must be called with a collection of some kind, (0, 1] was passed

In [6]: pd.Series(pd.Period("2019Q1", freq="Q"), index=range(3), dtype=pd.PeriodDtype("Q"))
---------------------------------------------------------------------------
ValueError: Buffer has wrong number of dimensions (expected 1, got 0)

In [7]: pd.Series(pd.Timestamp("2019", tz="US/Eastern"), index=range(3), dtype=pd.DatetimeTZDtype(tz="US/Eastern"))
---------------------------------------------------------------------------
TypeError: 'Timestamp' object is not iterable
```

Both of the above patterns appear to be working fine when using scalars that correspond to non-extensions dtypes (e.g. numeric, tz-naive, timedelta).

#### Problem description
The `Series` constructor is not correctly inferring the dtype from scalar `Interval`/`Period` objects when a dtype isn't specified, and raises for `Interval`/`Period`/tz-aware `Timestamp` when a dtype is specified.

#### Expected Output
I'd expect the `Series` to be constructed with the proper dtype.

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : 261c3a6673304f2f59f7cb2c1b285c3bdd2ff92e
python           : 3.7.3.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.14-041914-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 0.25.0+332.g261c3a667
numpy            : 1.16.4
pytz             : 2019.1
dateutil         : 2.8.0
pip              : 19.1.1
setuptools       : 41.0.1
Cython           : 0.29.10
pytest           : 4.6.2
hypothesis       : 4.23.6
sphinx           : 1.8.5
blosc            : None
feather          : None
xlsxwriter       : 1.1.8
lxml.etree       : 4.3.3
html5lib         : 1.0.1
pymysql          : None
psycopg2         : None
jinja2           : 2.10.1
IPython          : 7.5.0
pandas_datareader: None
bs4              : 4.7.1
bottleneck       : 1.2.1
fastparquet      : 0.3.0
gcsfs            : None
lxml.etree       : 4.3.3
matplotlib       : 3.1.0
numexpr          : 2.6.9
odfpy            : None
openpyxl         : 2.6.2
pandas_gbq       : None
pyarrow          : 0.11.1
pytables         : None
s3fs             : 0.2.1
scipy            : 1.2.1
sqlalchemy       : 1.3.4
tables           : 3.5.2
xarray           : 0.12.1
xlrd             : 1.2.0
xlwt             : 1.3.0
xlsxwriter       : 1.1.8
</details>

### PR DESCRIPTION
**Pull request by arw2019**
- [x] closes #28401
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jbrockmendel** [comment]
this looks fine.  should there be an analogous test in tests.extension?

**Comment by arw2019** [comment]
> this looks fine. should there be an analogous test in tests.extension?

Hmmm do you mean something like this?
``` python
In [7]: pd.array(pd.Interval(0, 1))
(...)
ValueError: Cannot pass scalar '(0, 1]' to 'pandas.array'.
```
We test this here:
https://github.com/pandas-dev/pandas/blob/52a1725965cb469de618ad56af467dc0aba53a87/pandas/tests/arrays/test_array.py#L287-L290
and 
https://github.com/pandas-dev/pandas/blob/52a1725965cb469de618ad56af467dc0aba53a87/pandas/tests/arrays/floating/test_construction.py#L85-L108

File: pandas/tests/series/test_constructors.py [hunk 1]
```diff
@@ -95,6 +95,14 @@ def test_scalar_conversion(self):
         assert float(Series([1.0])) == 1.0
         assert int(Series([1.0])) == 1
 
+    @pytest.mark.parametrize("scalar", (Interval(0, 1), Period("2019Q1", freq="Q")))
```
Comment (jreback) [inline pandas/tests/series/test_constructors.py:98]: can you create a fixture of the ea_scalars and use it here: https://github.com/pandas-dev/pandas/blob/master/pandas/tests/frame/test_constructors.py#L738 and in the new test you are adding

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_constructors.py (hunk 1 above)
Comment (jreback) [reply pandas/tests/series/test_constructors.py:98]: there are also similar tests for a dict of a scalar in both series/frame which could be updated to use the fixture as well

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_constructors.py (hunk 1 above)
Comment (arw2019) [reply pandas/tests/series/test_constructors.py]: > can you create a fixture of the ea_scalars and use it here: https://github.com/pandas-dev/pandas/blob/master/pandas/tests/frame/test_constructors.py#L738 and in the new test you are adding

Done

File: pandas/conftest.py [hunk 2]
```diff
@@ -687,6 +689,24 @@ def float_frame():
...
+# ----------------------------------------------------------------
+@pytest.fixture(
+    params=[
+        (Interval(0, 1), "interval[int64]"),
+        (Period("2019Q1", freq="Q"), "period[Q-DEC]"),
+        (
+            Timestamp("2011-01-01", tz="US/Eastern"),
+            DatetimeTZDtype(tz="US/Eastern"),
+        ),
+        (Timedelta(seconds=500), "timedelta64[ns]"),
```
Comment (arw2019) [inline pandas/conftest.py:703]: xref #38032 for a weird bug when the timedelta is in `ns`

Comment (jreback) [reply pandas/conftest.py:705]: great thanks. testing catching edge cases, who would think!

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @arw2019 

FYI there maybe other places where we can use this fixture, if its simple, would take a PR, if not, pls open an issue.

**Comment by arw2019** [comment]
> FYI there maybe other places where we can use this fixture, if its simple, would take a PR, if not, pls open an issue.

Working on it. Probably will be a single PR but will see

**Pull request merged**